ports/espressif/bindings/espnow/ESPNow.c

```diff
@@ -0,0 +1,876 @@
+    self->tx_packets += ((peer_addr == NULL) ? self->peers_count : 1);
+
+    if (sync) {
+        // Wait for and tally all the expected responses from peers
+        _wait_for_pending_responses(self);
+    }
+
+    // Return False if sync and any peers did not respond.
+    return mp_obj_new_bool(!(sync && self->tx_failures != saved_failures));
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(espnow_send_obj, 2, espnow_send);
+
+//|     def recv(self, buffers: List[bytearray]) -> int:
+//|         """Loads mac, message, rssi and timestamp into the provided buffers.
+//|
+//|         If buffers is 2 elements long, the mac and message will be
+//|         loaded into the 1st and 2nd elements.
+//|         If buffers is 4 elements long, the rssi and timestamp values will be
+//|         loaded into the 3rd and 4th elements.
+//|
+//|         :param List[bytearray] buffers: List of buffers to be loaded.
+//|
+//|         :returns: Length of the message."""
+//|         ...
+STATIC mp_obj_t espnow_recv(mp_obj_t self_in, mp_obj_t buffers) {
+    espnow_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+
+    mp_obj_list_t *list = MP_OBJ_TO_PTR(buffers);
+    if (!mp_obj_is_type(list, &mp_type_list) || list->len < 2) {
+        mp_arg_error_invalid(MP_QSTR_buffers);
+    }
+
+    mp_obj_array_t *msg = MP_OBJ_TO_PTR(list->items[1]);
+    if (mp_obj_is_type(msg, &mp_type_bytearray)) {
+        msg->len += msg->free;   // Make all the space in msg array available
+        msg->free = 0;
+    }
+
+    uint8_t *peer_buf = _get_bytes_len(list->items[0], ESP_NOW_ETH_ALEN, MP_BUFFER_WRITE);
+    uint8_t *msg_buf = _get_bytes_len(msg, ESP_NOW_MAX_DATA_LEN, MP_BUFFER_WRITE);
+
+    // Read the packet header from the incoming buffer
+    espnow_header_t header;
+    if (!ringbuf_get_n(self->recv_buffer, (uint8_t *)&header, sizeof(header))) {
+        return MP_OBJ_NEW_SMALL_INT(0);
+    }
+
+    uint8_t msg_len = header.msg_len;
+
+    // Check the message packet header format and read the message data
+    if (header.magic != ESPNOW_MAGIC ||
+        msg_len > ESP_NOW_MAX_DATA_LEN ||
+        !ringbuf_get_n(self->recv_buffer, peer_buf, ESP_NOW_ETH_ALEN) ||
+        !ringbuf_get_n(self->recv_buffer, msg_buf, msg_len)) {
+        mp_arg_error_invalid(MP_QSTR_buffer);
+    }
+    if (mp_obj_is_type(msg, &mp_type_bytearray)) {
+        // Set the length of the message bytearray.
+        size_t size = msg->len + msg->free;
+        msg->len = msg_len;
+        msg->free = size - msg_len;
+    }
+
+    // Update rssi value in the peer device table
+    _update_rssi(self, peer_buf, header.rssi, header.time_ms);
+    if (list->len == 4) {
+        list->items[2] = MP_OBJ_NEW_SMALL_INT(header.rssi);
+        list->items[3] = mp_obj_new_int(header.time_ms);
+    }
+
+    return MP_OBJ_NEW_SMALL_INT(msg_len);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(espnow_recv_obj, espnow_recv);
+
+//|     any: bool
+//|     """`True` if data is available to read from the buffers."""
```

Comment:
Maybe fold this into the truthiness of the object itself via unary_op. Then you could do `if enow:`